Tests/SpeziFoundationTests/ManagedAsynchronousAccessTests.swift

```diff
@@ -6,260 +6,269 @@
-
-        await fulfillment(of: [expectation], timeout: 2.0)
     }
-
+    
     @MainActor
+    @Test
     func testCancelAll() async throws {
         let access = ManagedAsynchronousAccess<Void, Error>()
 
-        let expectation = XCTestExpectation(description: "task")
-        let handle = Task {
-            do {
-                _ = try await access.perform {}
-                XCTFail("Expected cancellation error.")
-            } catch {
-                XCTAssertTrue(error is CancellationError)
+        try await confirmation("perform() returns with cancellation error") { confirm in
+            let task = Task {
+                do {
+                    _ = try await access.perform {}
+                    Issue.record("Expected cancellation error.")
+                } catch {
+                    #expect(error is CancellationError)
+                }
+                confirm()
             }
-            expectation.fulfill()
+            
+            try await Task.sleep(for: .milliseconds(100))
+            #expect(access.ongoingAccess == true)
+            #expect(task.isCancelled == false)
+            
+            access.cancelAll()
+
+            #expect(access.ongoingAccess == false)
+            
+            await task.value
+            
+            #expect(task.isCancelled, "Task should be marked as cancelled.")
         }
-
-        try await Task.sleep(for: .milliseconds(100))
-
-        XCTAssertTrue(access.ongoingAccess)
-        XCTAssertFalse(handle.isCancelled)
-
-        access.cancelAll()
-
-        XCTAssertFalse(access.ongoingAccess)
-        await fulfillment(of: [expectation])
-
-        XCTAssert(handle.isCancelled, "Task was not marked as cancelled.")
     }
-
+    
     @MainActor
+    @Test
     func testCancelAllNeverError() async throws {
         let access = ManagedAsynchronousAccess<Void, Never>()
 
-        let expectation = XCTestExpectation(description: "task")
-        let handle = Task {
-            do {
-                try await access.perform {}
-                XCTFail("Expected cancellation to turn into a cancellation error")
-            } catch {
-                XCTAssertTrue(error is CancellationError)
+        try await confirmation("perform() returns with cancellation error") { confirm in
+            let task = Task {
+                do {
+                    try await access.perform {}
+                    Issue.record("Expected cancellation to turn into a cancellation error")
+                } catch {
+                    #expect(error is CancellationError)
+                }
+                confirm()
             }
-            expectation.fulfill()
+            
+            try await Task.sleep(for: .milliseconds(100))
+            
+            #expect(access.ongoingAccess == true)
+            #expect(task.isCancelled == false)
+            
+            access.cancelAll()
+            
+            #expect(access.ongoingAccess == false)
```

Comment:
nit (here and in a couple of other places): we should probably remove the `== true` and write `!x` instead of `x == false`.